pkgs/top-level/all-packages.nix

```diff
@@ -16951,6 +16951,8 @@ in
 
   starship = callPackage ../tools/misc/starship { };
 
+  stig = callPackage ../applications/networking/p2p/stig { python = python3; };
```

Comment:
Ah, this somewhat negates my previous point about python2 vs. python3. It is surprising to people reading only the `default.nix` though, since `python` usually refers to python2. It would be better to explicitly take `python3` as an argument.

Reply:
The following is my response to both this and your [first comment](https://github.com/NixOS/nixpkgs/pull/67233/files/e1de8a6aaa368f5e6ab79202d31f1f93191a1096#r317353800):

TBH, with all due respect, the whole Python2/3 issue isn't handled very good IMO in Nixpkgs. Python2 is going to retire in 1/1/2020 (!) and no body is writing anything new in it and every active software that uses it usually strives to rewrite their apps in Python 3. Python 3 is also the default Python version on many other distributions (when they refer to `python`).

Therefor, my opinion is that we should use the `python` argument in the packages' expressions and use `python = python3` in the `callPackage` statements in `pkgs/top-level/` files. I think that in the hopefully near future, when we'll finally remove python2 or at least make `python` refer to `python3` and not `python2`, we'll be able to easily iterate over only the `top-level` expressions and thus everything will be a little bit more consistent.

Reply:
I will point you to https://github.com/NixOS/nixpkgs/issues/18185, individual pull requests are not a place to try to change policy.

Reply:
I wasn't by any means trying to be too critical. it's just seems to me, that as for the 2nd TODO item in [`@domenkozar`'s first comment in that issue](https://github.com/NixOS/nixpkgs/issues/18185#issue-174489897), it'll be easier to spot packages that use python3 by default and that need no `python = python3` by looking at a single file - `pkgs/top-level/all-packages.nix`.

What will happen when Python 4 will be out? Assuming (hopefully) it won't include any breaking changes for Python 3 programs, we will be able to switch easily to it for all packages that use just `python`.

I assume this is very much out of scope for this review. I'm just thinking out loud, and TBH, I didn't read every comment in [that Python 3 migration issue](https://github.com/NixOS/nixpkgs/issues/18185). If the policy is to use `python3` in packages' expressions, then I'll use that. I've had a similar discussion in my [PR for improving `sequoia`](https://github.com/NixOS/nixpkgs/pull/65724#discussion_r310132597) but there we used eventually the `python* = python3*` scheme in the `pkgs/top-level/` files.

Reply:
Definitely not trying to discourage criticism :) I get your point, although I'm not sure I agree that this is a good time for it. The official deprecation being so close, we can just do the default transition afterwards.

Anyway I think python2/python3 is a very special case, that will not be repeated. I'd be very surprised if anyone would make the same mistake with python4. I'd even be surprised if there ever *was* a python4, due to the negative connotations alone.

But the fact is that currently python2 *is* the default, and changing that in a single expression is just going to be confusing for little gain. If we want to switch, we can just grep for `python3`. If you feel strongly about it, you can make your case in the migration issue or on discord and see if other people (especially the main python maintainer FRidth) agree and then work on changing the default.

Reply:
Oh right then. I've pushed a final change I hope you'll approve. I count on `@FRidh` to do the right thing on time so I won't bother to mention my small opinion on the matter.

Thanks for your patience and attentiveness. I'm marking this conversation as resolved.